src/main/java/de/dennisguse/opentracks/io/file/post_workout_export/InstantPostWorkoutExport.java

```diff
@@ -0,0 +1,60 @@
+package de.dennisguse.opentracks.io.file.post_workout_export;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.documentfile.provider.DocumentFile;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
+import de.dennisguse.opentracks.io.file.TrackFileFormat;
+import de.dennisguse.opentracks.util.StringUtils;
+
+public class InstantPostWorkoutExport {
+
+    private static final String TAG = InstantPostWorkoutExport.class.getSimpleName();
+
+    private Context context;
+    private ContentProviderUtils contentProviderUtils;
+    private SharedPreferences sharedPreferences;
+    private TrackFileFormat trackFileFormat;
+
+    public InstantPostWorkoutExport(Context context, ContentProviderUtils contentProviderUtils, SharedPreferences sharedPreferences, TrackFileFormat trackFileFormat) {
+        this.context = context;
+        this.contentProviderUtils = contentProviderUtils;
+        this.sharedPreferences = sharedPreferences;
+        this.trackFileFormat = trackFileFormat;
+    }
+
+    public void exportLastTrackToExternalStorage(String uriAsString) {
+        String directoryAsString = sharedPreferences.getString(uriAsString, "invalid uri");
+        Uri directoryUri = Uri.parse(directoryAsString);
+        DocumentFile directory = DocumentFile.fromTreeUri(context, directoryUri);
+        assert directory != null;
```

Comment:
Not necessary as we would die in line 45 with an NPE anyhow:
```
         DocumentFile file = directory.findFile(fileName);
```

Reply:
android studio suggested this. i removed it now